tools/scripts/failure-event-handler

```diff
@@ -2,24 +2,75 @@
 import sys
 import os
 import signal
+import datetime
+
+from datetime import timezone
+
 
 def write_stdout(s):
     sys.stdout.write(s)
     sys.stdout.flush()
 
+
 def write_stderr(s):
     sys.stderr.write(s)
     sys.stderr.flush()
 
+
 def main():
     while 1:
-        write_stdout('READY\n')
+        write_stdout("READY\n")
+
+        # read header line and print it to stderr
         line = sys.stdin.readline()
-        headers = dict([ x.split(':') for x in line.split() ])
-        headers.update(dict([ x.split(':') for x in sys.stdin.read(int(headers['len'])).split()]))
-        if headers['eventname'] == 'PROCESS_STATE_FATAL':
-            os.kill(os.getppid(), signal.SIGTERM)
-        write_stdout('RESULT 2\nOK')
+
+        headers = None
+        data = None
+
+        # parse header line and decide what to do with it
+        try:
+            headers = dict([x.split(":") for x in line.split()])
+            data = sys.stdin.read(int(headers["len"]))
+        except ValueError as e:
+            write_stderr(str(e))
+
+        # now decide what do to based on eventnames
+        if headers["eventname"] == "PROCESS_STATE_FATAL":
+            headers.update(
+                dict(
+                    [x.split(":") for x in sys.stdin.read(int(headers["len"])).split()]
+                )
+            )
+
+            try:
+                # incoming event that produced PROCESS_STATE_FATAL will have a PID. SIGTERM it!
+                write_stderr(
+                    f"{datetime.datetime.now(timezone.utc)} - sending SIGTERM to proc={headers} with data={headers}\n"
+                )
+                os.kill(headers["pid"], signal.SIGTERM)
```

Comment:
So is this a change in behavior from before when it was

```python
os.kill(os.getppid(), signal.SIGTERM)
```

It's true that yours looks better, but it causes me some concerns about the exception cases. If the headers doesn't have a "len" value, but the "eventname" is PROCESS_STATE_FATAL, maybe we really do want to SIGTERM `os.getppid()`